plugin/proc_info/proc_info.cc

```diff
@@ -0,0 +1,177 @@
+#include <my_global.h>
+#include <table.h>
+#include <sql_show.h>
+#include <sql_class.h>
+#include <mysql/plugin.h>
+
+/* We'll use this to read directories */
+#include <dirent.h>
+
+
+static ST_FIELD_INFO meminfo_fields[]=
+{
+  { "NAME", 100, MYSQL_TYPE_STRING, 0, 0, 0, SKIP_OPEN_TABLE },
+  { "VALUE", 21, MYSQL_TYPE_LONG, 0, MY_I_S_UNSIGNED, 0, SKIP_OPEN_TABLE },
```

Comment:
MYSQL_TYPE_LONG is int32. Probably not sufficient to describe virtual memory, even in kilobytes. I think MYYSQL_TYPE_LONGLONG is more appropriate. also  please consider a longlong C datatype when parsing meminfo later when filling table.